File: src/main/java/com/danielpark/imagecropper/util/log/Logger.java
package com.danielpark.imagecropper.util.log;

import android.util.Log;

/**
 * Copyright (C) 2014-2016 daniel@bapul.net
 * Created by Daniel on 2016-12-07.
 */

public class Logger {

    private static Logger sThis;

    private Logger(){}

    public synchronized static final Logger getInstance() {
        if (sThis == null)
            sThis = new Logger();

        return sThis;
    }

    private final String TAG = "ImageCropperLogger";
    private boolean mLogFlag = ImageCropperLogger.getLogState();

    public void v(String msg) {
        if (mLogFlag) {
            Log.v(TAG, "" + msg);
        }
    }

    public void d(String msg) {
        if (mLogFlag) {
            Log.d(TAG, "" + msg);
        }
    }

    public void e(String msg) {
        if (mLogFlag) {
            Log.e(TAG, "" + msg);
        }
    }

    public void i(String msg) {
        if (mLogFlag) {
            Log.i(TAG, "" + msg);
        }
    }

    public void w(String msg) {
        if (mLogFlag) {
            Log.w(TAG, "" + msg);
        }
    }

    public void wtf(String msg) {
        if (mLogFlag) {
            Log.wtf(TAG, "" + msg);
        }
    }
}

File: library/src/main/java/com/danielpark/imagecropper/util/Logger.kt
package com.danielpark.imagecropper.util

import android.util.Log
import com.danielpark.imagecropper.BuildConfig

/**
 * print log with specific information.
 * <br><br>
 * Created by Daniel Park on 2019-12-18
 */
object Logger {

    private val ENABLE_LOGGER : Boolean = BuildConfig.DEBUG

    private fun buildLogMsg(message: String): String {
        val ste = Thread.currentThread().stackTrace[4]
        val sb = StringBuilder()
        sb.append("(thread : ")
        sb.append(Thread.currentThread().name)
        sb.append(")")
        sb.append("[[")
        sb.append(ste.fileName)
        sb.append(">")
        sb.append(ste.methodName)
        sb.append(">#")
        sb.append(ste.lineNumber)
        sb.append("]] ")
        sb.append(message)
        return sb.toString()
    }

    fun v(tag: String, msg: String) {
        if (ENABLE_LOGGER) {
            Log.v(tag, buildLogMsg(msg))
        }

    }

    fun d(tag: String, msg: String) {
        if (ENABLE_LOGGER) {
            Log.d(tag, buildLogMsg(msg))
        }

    }

    fun w(tag: String, msg: String) {
        if (ENABLE_LOGGER) {
            Log.w(tag, buildLogMsg(msg))
        }

    }

    fun w(tag: String, msg: String, throwable: Throwable) {
        if (ENABLE_LOGGER) {
            Log.w(tag, buildLogMsg(msg), throwable)
        }

    }

    fun w(tag: String, throwable: Throwable) {
        if (ENABLE_LOGGER) {
            Log.w(tag, throwable)
        }

    }

    fun e(tag: String, msg: String) {
        if (ENABLE_LOGGER) {
            Log.e(tag, buildLogMsg(msg))
        }

    }

    fun e(tag: String, msg: String, throwable: Throwable) {
        if (ENABLE_LOGGER) {
            Log.e(tag, buildLogMsg(msg), throwable)
        }

    }

    fun e(tag: String, throwable: Throwable) {
        if (ENABLE_LOGGER) {
            Log.e(tag, "", throwable)
        }

    }

    fun i(tag: String, msg: String) {
        if (ENABLE_LOGGER) {
            Log.i(tag, buildLogMsg(msg))
        }

    }
}